src/tqec/compile/detectors/database.py

```diff
@@ -104,8 +108,28 @@ def plaquette_names(self) -> tuple[tuple[tuple[str, ...], ...], ...]:
 
     @cached_property
     def reliable_hash(self) -> int:
-        """Returns a hash of `self` that is guaranteed to be constant across
-        Python versions, OSes and executions.
+        """Returns a hash of ``self`` that is guaranteed to be constant.
+
+        Python's ``hash`` is not guaranteed to be constant across Python versions, OSes and
+        executions. In particular, strings hash will not be repeatable across different Python
+        executable calls. The following line should return different values at each call:
+
+        .. code:: bash
+
+            python -c 'print(hash("Hello world"))'
+
+        For example, here are the results obtained after calling that line 3 times on my machine:
+
+        - ``2656127643635015930``
+        - ``1413792191058799258``
+        - ``8731178165517315210``
+
+        This is an issue for the detector database as we would like it to be reproducible. Else,
+        re-using an existing database will always fail because keys will have a different hash,
+        hence re-computing detectors at each calls and growing indefinitely the database.
```

Comment:
```suggestion
        reusing an existing database will always fail because keys will have a different hash,
        hence re-computing detectors at each call and growing the database indefinitely.
```